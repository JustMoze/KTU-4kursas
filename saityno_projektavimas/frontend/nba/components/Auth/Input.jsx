import React, { useState } from 'react'
import TextField from '@material-ui/core/TextField'
import styled from 'styled-components'

function Input({ handleChange, error, message, name, ...props }) {
  const WhiteBorderTextField = styled(TextField)`
    & label.Mui-focused {
      color: ${error ? 'red' : 'green'};
    }
    & label.Mui {
      color: ${error ? 'red' : 'green'};
    }
    & .MuiOutlinedInput-root {
      &.Mui-focused fieldset {
        border-color: ${error ? 'red' : 'green'};
      },
    }
    & .MuiOutlinedInput-root {
      & fieldset {
        border-color: ${error ? 'red' : 'green'};
      },
    }

    & p.MuiFormHelperText-root {
            color: ${error ? 'red' : '#323232'}
        }
    }
  `
  const ConfigureName = (name) => {
    return name.charAt(0).toUpperCase() + name.slice(1)
  }
  return (
    <WhiteBorderTextField
      error={error}
      label={ConfigureName(name)}
      helperText={error && message}
      variant="outlined"
      onChange={(text) => handleChange(text)}
      name={name}
      style={{
        width: '100%',
        marginBottom: 10,
      }}
      {...props}
    />
  )
}

export default Input
