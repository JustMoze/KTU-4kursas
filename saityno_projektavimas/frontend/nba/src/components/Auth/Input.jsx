import React from 'react'
import TextField from '@material-ui/core/TextField'
import styled from 'styled-components'

export const WhiteBorderTextField = styled(TextField)`
  & label.Mui-focused {
    color: ${(props) => (props.error ? 'red' : props.color)};
  }
  & label.Mui {
    color: ${(props) => (props.error ? 'red' : props.color)};
  }
  & .MuiOutlinedInput-root {
    &.Mui-focused fieldset {
      border-color: ${(props) => (props.error ? 'red' : props.color)};
    },
  }
  & .MuiOutlinedInput-root {
    & fieldset {
      border-color: ${(props) => (props.error ? 'red' : props.color)};
    },
  }

  & p.MuiFormHelperText-root {
          color: ${(props) => (props.error ? 'red' : '#323232')}
      }
  }
`
function Input({ handleChange, error, message, name, color, ...props }) {
  const ConfigureName = (name) => {
    name = name.slice(name.indexOf('_') + 1, name.length)
    return name.charAt(0).toUpperCase() + name.slice(1)
  }
  return (
    <WhiteBorderTextField
      color={color}
      error={error}
      label={ConfigureName(name)}
      helperText={error && message}
      variant="outlined"
      onChange={(text) => handleChange(text)}
      name={name}
      style={{
        width: '100%',
        marginBottom: 10,
      }}
      {...props}
    />
  )
}

export default Input
