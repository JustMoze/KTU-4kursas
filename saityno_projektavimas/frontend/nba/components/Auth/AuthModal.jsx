import React, { useState, useEffect } from 'react'
import Button from '@material-ui/core/Button'
import Modal from '@material-ui/core/Modal'
import Input from './Input'
import styled from 'styled-components'
import { makeStyles } from '@material-ui/core/styles'
import { RiLoginCircleLine } from 'react-icons/ri'
import { AiFillCloseCircle } from 'react-icons/ai'
import { NBA_blue } from '../../styles/globalStyle.module.scss'
import { login, register } from '../../utils/Inputs'

const useStyles = makeStyles((theme) => ({
  button: {
    margin: theme.spacing(1),
    width: '70%',
    borderRadius: 30,
    backgroundColor: NBA_blue,
    '&:hover, &:focus': {
      width: '80%',
      backgroundColor: 'green',
      minHeight: 40,
    },
  },
}))

// const InputContainer = styled.div`
//   width: 100%;
//   padding-right: 20px;
//   padding-left: 20px;
// `

const ModalContainer = styled.div`
  width: 100%;
  height: 100%;
  justify-content: center;
  align-items: center;
  background-color: white;
  position: relative;
  border-radius: 10px;
`
const CenterContainer = styled.div`
  padding-top: 30px;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 90%;
`
const CustomModal = styled(Modal)`
  position: absolute;
  width: 40vw;
  height: 60vh;
  min-height: 500px;
  zindex: 10;
  padding: 10px;
  border-radius: 10px;
  justify-content: center;
  align-items: center;
`
const IconContainer = styled.div`
  position: absolute;
  top: 5px;
  cursor: pointer;
  right: 5px;
  min-width: 25px;
  min-height: 25px;
`

const AuthModal = ({ open, handleClose, type }) => {
  const classes = useStyles()
  const [allowSubmit, setAllowSubmit] = useState(false)
  const inputs = type.toLowerCase() === 'register' ? register : login
  const [user, setUser] = useState({})

  const ValidateInput = (input) => {
    return false
  }

  const HandleUserCredentialChange = (e) => {
    let { value, name } = e.target
    setUser({ ...user, [name]: value })
  }

  return (
    <CustomModal
      open={open}
      onBackdropClick={handleClose}
      onEscapeKeyDown={handleClose}
      onClose={handleClose}
      style={{
        top: type.toLowerCase() === 'register' ? '10vh' : '25vh',
        left: '30vw',
        minHeight: type.toLowerCase() === 'register' ? '80vh' : 400,
      }}
    >
      <ModalContainer>
        <IconContainer onClick={handleClose}>
          <AiFillCloseCircle size={25} />
        </IconContainer>
        <CenterContainer>
          {inputs.map((input, index) => {
            return (
              <Input
                key={index}
                error={ValidateInput(input)}
                message="Need to fill it"
                name={input.name}
                value={user[`${input.name}`]}
                handleChange={(e) => HandleUserCredentialChange(e)}
                type={input.type}
              />
            )
          })}
          <div
            style={{
              width: '100%',
              marginTop: 20,
              display: 'flex',
              justifyContent: 'center',
            }}
          >
            <Button
              variant="contained"
              size="small"
              className={classes.button}
              disabled={!allowSubmit}
              startIcon={<RiLoginCircleLine size={25} />}
            >
              Save
            </Button>
          </div>
        </CenterContainer>
      </ModalContainer>
    </CustomModal>
  )
}

export default AuthModal
