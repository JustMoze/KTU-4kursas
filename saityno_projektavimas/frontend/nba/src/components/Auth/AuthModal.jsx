import React from 'react'
import Button from '@material-ui/core/Button'
import Modal from '@material-ui/core/Modal'
import Input from './Input'
import styled from 'styled-components'
import { makeStyles } from '@material-ui/core/styles'
import { RiLoginCircleLine } from 'react-icons/ri'
import { AiFillCloseCircle } from 'react-icons/ai'
import { login, register, validateLength } from '../../../utils/Inputs'
import has from 'lodash/has'
import { validateEmail } from './../../../utils/Inputs'

const useStyles = makeStyles((theme) => ({
  button: {
    margin: theme.spacing(1),
    width: '70%',
    borderRadius: 30,
    backgroundColor: '#1e438a',
    '&:hover, &:focus': {
      width: '80%',
      backgroundColor: 'green',
      minHeight: 40,
    },
  },
}))
export const LogoContainer = styled.div`
  background-color: ${(props) => props.color};
  width: 30%;
  height: 60px;
  justify-content: center;
  align-self: center;
  align-items: center;
  display: flex;
  margin-bottom: 25px;
  padding: 3px;
  border-radius: 15px;
`
const ModalContainer = styled.div`
  width: 100%;
  height: 100%;
  justify-content: center;
  align-items: center;
  background-color: white;
  position: relative;
  border-radius: 10px;
`
const CenterContainer = styled.div`
  padding-top: 30px;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 90%;
`
const CustomModal = styled(Modal)`
  position: absolute;
  background-color: #d4d7dd;
  width: 40vw;
  height: 60vh;
  min-height: 500px;
  zindex: 10;
  border-radius: 10px;
  justify-content: center;
  align-items: center;
`
const IconContainer = styled.div`
  position: absolute;
  top: 5px;
  cursor: pointer;
  right: 5px;
  min-width: 25px;
  min-height: 25px;
`

const AuthModal = ({
  open,
  handleClose,
  type,
  user,
  handleChange,
  handleSubmit,
  color,
}) => {
  const classes = useStyles()
  const inputs = type.toLowerCase() === 'register' ? register : login
  var errArr = {}

  const ValidateInput = (input) => {
    let { validation } = input
    if (has(validation, 'email')) {
      if (!validateEmail(user[`${input.name}`])) {
        errArr[`${input.name}`] = 1
        return 'Email does not match format'
      } else {
        errArr[`${input.name}`] = 0
      }
    } else if (has(validation, 'min') && has(validation, 'max')) {
      let { min, max } = validation
      let errArrs = validateLength(min, max, user[`${input.name}`], input.name)
      if (errArrs.length > 0) {
        errArr[`${input.name}`] = 1
        return errArrs[0]
      } else {
        errArr[`${input.name}`] = 0
      }
    }
    if (errArr[`${input.name}`] === 0) {
      return ''
    }
  }
  const AllowSubmit = () => {
    let counter = 0
    for (let prop in errArr) {
      if (errArr[prop] === 0) {
        counter++
      }
    }
    if (counter === inputs.length) {
      return true
    } else return false
  }
  return (
    <CustomModal
      open={open}
      onBackdropClick={handleClose}
      onEscapeKeyDown={handleClose}
      onClose={handleClose}
      style={{
        top: type.toLowerCase() === 'register' ? '10vh' : '25vh',
        left: '30vw',
        minHeight: type.toLowerCase() === 'register' ? '80vh' : 400,
      }}
    >
      <ModalContainer>
        <IconContainer id="iconContainer" onClick={handleClose}>
          <AiFillCloseCircle size={25} />
        </IconContainer>
        <CenterContainer>
          <div
            style={{
              width: '100%',
              display: 'flex',
              alignSelf: 'center',
              justifyContent: 'center',
            }}
          >
            <LogoContainer color={color}>
              <img
                src="https://ak-static.cms.nba.com/wp-content/uploads/logos/leagues/logo-nba.svg"
                height="100%"
              />
            </LogoContainer>
          </div>
          {inputs.map((input, index) => {
            return (
              <Input
                color={color}
                key={index}
                error={ValidateInput(input) === '' ? false : true}
                message={ValidateInput(input)}
                name={`${type}_${input.name}`}
                value={user[`${input.name}`]}
                handleChange={(e) => handleChange(e)}
                type={input.type}
              />
            )
          })}
          <div
            style={{
              width: '100%',
              marginTop: 20,
              display: 'flex',
              justifyContent: 'center',
            }}
          >
            <Button
              variant="contained"
              size="small"
              style={{ backgroundColor: color, color: '#ffffff' }}
              className={classes.button}
              onClick={() => handleSubmit(type)}
              disabled={!AllowSubmit()}
              startIcon={<RiLoginCircleLine size={25} />}
            >
              Save
            </Button>
          </div>
        </CenterContainer>
      </ModalContainer>
    </CustomModal>
  )
}

export default AuthModal
