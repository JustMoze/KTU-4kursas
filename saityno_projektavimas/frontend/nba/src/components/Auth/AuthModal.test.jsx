import React from 'react'
import { shallow } from 'enzyme'
import AuthModal, { LogoContainer } from './AuthModal'
import toJson from 'enzyme-to-json'
import Input from './Input'
import 'jest-styled-components'


describe('Auth modal', () => {
	const handleClose = jest.fn();
	const handleChange = jest.fn();
	const handleSubmit = jest.fn();

    const initProps = {
        color: '#ffffff',
        open: true,
        user: {
            email: 'kompotas@gmail.com', password: 'kompotas123'
        },
        type: 'login', 
        handleClose: handleClose,
        handleChange: handleChange,
        handleSubmit: handleSubmit
    }
    const wrapper = shallow(<AuthModal {...initProps} />);

    it('AuthModal renders correctly', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('should render styled component', () => {
        const styledWrapper = shallow(<LogoContainer color={initProps.color} />)
        let toJsonComponent = JSON.stringify(styledWrapper.debug());
        let indexOf = toJsonComponent.indexOf('color') + 7;
        let color = toJsonComponent.slice(indexOf, indexOf+8);
        let fullColor = color.concat("\"");
        expect(fullColor).toMatch(initProps.color);
    });

    it('renders all Auth modal input components', () => {
		expect(wrapper.find(Input)).toHaveLength(2);
	});

	afterEach(() => {
		jest.clearAllMocks();
	});

	describe('handle close', () => {
		it('handle close is called after click', () => {
			wrapper.find('#iconContainer').simulate('click');
			expect(handleClose).toHaveBeenCalled();
		});
	});
})