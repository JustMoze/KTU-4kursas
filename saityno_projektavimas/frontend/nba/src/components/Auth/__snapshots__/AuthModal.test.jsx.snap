// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth modal AuthModal renders correctly 1`] = `
<AuthModal__CustomModal
  onBackdropClick={[MockFunction]}
  onClose={[MockFunction]}
  onEscapeKeyDown={[MockFunction]}
  open={true}
  style={
    Object {
      "left": "30vw",
      "minHeight": 400,
      "top": "25vh",
    }
  }
>
  <AuthModal__ModalContainer>
    <AuthModal__IconContainer
      id="iconContainer"
      onClick={[MockFunction]}
    >
      <AiFillCloseCircle
        size={25}
      />
    </AuthModal__IconContainer>
    <AuthModal__CenterContainer>
      <div
        style={
          Object {
            "alignSelf": "center",
            "display": "flex",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <AuthModal__LogoContainer
          color="#ffffff"
        >
          <img
            height="100%"
            src="https://ak-static.cms.nba.com/wp-content/uploads/logos/leagues/logo-nba.svg"
          />
        </AuthModal__LogoContainer>
      </div>
      <Input
        color="#ffffff"
        error={false}
        handleChange={[Function]}
        key="0"
        message=""
        name="login_email"
        type="email"
        value="kompotas@gmail.com"
      />
      <Input
        color="#ffffff"
        error={false}
        handleChange={[Function]}
        key="1"
        message=""
        name="login_password"
        type="password"
        value="kompotas123"
      />
      <div
        style={
          Object {
            "display": "flex",
            "justifyContent": "center",
            "marginTop": 20,
            "width": "100%",
          }
        }
      >
        <WithStyles(ForwardRef(Button))
          className="makeStyles-button-1"
          disabled={false}
          onClick={[Function]}
          size="small"
          startIcon={
            <RiLoginCircleLine
              size={25}
            />
          }
          style={
            Object {
              "backgroundColor": "#ffffff",
              "color": "#ffffff",
            }
          }
          variant="contained"
        >
          Save
        </WithStyles(ForwardRef(Button))>
      </div>
    </AuthModal__CenterContainer>
  </AuthModal__ModalContainer>
</AuthModal__CustomModal>
`;
